src/job/MomoJob.ts

```diff
@@ -1,10 +1,53 @@
 export type Handler = () => Promise<string | undefined | void> | string | undefined | void;
 
-export interface MomoJob {
+interface MomoBaseJob {
   handler: Handler;
   name: string;
-  interval: string;
-  firstRunAfter?: number;
   concurrency?: number;
   maxRunning?: number;
 }
+
+export interface IntervalSchedule {
+  interval: string;
+  firstRunAfter: number;
+}
+
+export interface CronSchedule {
+  cronSchedule: string;
+}
+
+export function isIntervalSchedule(input: unknown): input is IntervalSchedule {
+  return (
+    // @ts-ignore
+    input.interval !== undefined &&
+    // @ts-ignore
+    input.firstRunAfter !== undefined &&
+    typeof (input as IntervalSchedule).interval == 'string' &&
+    typeof (input as IntervalSchedule).firstRunAfter == 'number'
```

Comment:
IntelliJ spits out warnings here and I also don't like the ts-ignores at all, which is why I would suggest something like

```
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isInterval(input: any): input is Interval {
  return (
    input.interval !== undefined &&
    input.firstRunAfter !== undefined &&
    typeof input.interval == 'string' &&
    typeof input.firstRunAfter == 'number'
  );
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isCronSchedule(input: any): input is CronSchedule {
  return input.cronSchedule !== undefined && typeof input.cronSchedule == 'string';
}
```
I would definitely prefer suppress an eslint warning over suppressing compile errors ...

Reply:
Yes, of course! This is much better.